### PR TITLE
Regression assessment

### PR DESCRIPTION
**Pull request by amrasekh**
This PR adds regression assessment metrics. Plus some updates to make it possible to run it. 

An example to try them:

```
from sklearn.linear_model import LinearRegression
import pandas as pd
import credoai.lens as cl
from random import uniform

df = pd.DataFrame(
    [['Male', 1.1, 2], ['Male', 2, 3], ['Male', 2.2, 1], ['Male', 2.8, 2], 
    ['Female', 3, 4], ['Female', 1, 4], ['Female', 5.1, 3]],
    columns=['SEX', 'AGE', 'EDUCATION']
    )

df['target'] = df.apply(lambda x: (2 * x.EDUCATION + 3 * x.AGE) * uniform(0.9, 1.1), axis=1)

X = df[['AGE', 'EDUCATION']]
y = df['target']

model = LinearRegression().fit(X, y)

credo_model = cl.CredoModel(name='income_predictor',
                            model=model, 
                            model_config={'pred_fun': model.predict})
                            
credo_data = cl.CredoData(name='income_dataset',
                          data=df,
                          sensitive_feature_key='SEX',
                          label_key='target',
                          categorical_features_keys=['SEX', 'EDUCATION']
                          )

metrics = ['mean_absolute_error', 'explained_variance_score', 'r2_score']

alignment_spec = {'Fairness': {'metrics': metrics},
                  'Performance': {'metrics': metrics}}

lens = cl.Lens(model=credo_model,
               data=credo_data,
               spec=alignment_spec)
               
lens.run_assessments().display_results()
```

Also came across and fixed a bug in DatasetFairness for when sensitive feature is continuous.

Next Steps:
* Adding regression-specific (continuous target) reporting (visualizations)

### CODE REVIEW

File: credoai/artifacts.py [hunk 1]
```diff
@@ -379,7 +379,7 @@ def __init__(
         self.name = name
         self.config = {}
         assert model is not None or model_config is not None
-        if model is not None:
+        if model is not None and model_config is None:
```
Comment (IanAtCredo) [inline credoai/artifacts.py:382]: What was this change for? The reason I had set it up like so was to make sure we are always inferring what we can from the model, and only updating what was explicitly asked for.

File: credoai/metrics/metric_constants.py [hunk 2]
```diff
@@ -62,3 +62,19 @@
     "demographic_parity_ratio",
     "demographic_parity_difference"
 ]
+
+# REGRESSION METRICS
+REGRESSION_FUNCTIONS = {
```
Comment (IanAtCredo) [inline credoai/metrics/metric_constants.py:67]: Can you add to `METRIC_EQUIVALENTS` above? For instance, we probably want to add things like:
```
'mean_absolute_error': ['MAE'],
'mean_squared_error': ['MSE'],
...
```

File: credoai/metrics/metric_constants.py [hunk 3]
```diff
@@ -62,3 +62,19 @@
...
+    'explained_variance_score': sk_metrics.explained_variance_score,
+    'max_error': sk_metrics.max_error,
+    'mean_absolute_error': sk_metrics.mean_absolute_error,
+    'mean_squared_error': sk_metrics.mean_squared_error,
```
Comment (IanAtCredo) [inline credoai/metrics/metric_constants.py:71]: Can you add an "RMSE" metric? I think it's common enough we'd want it. I think using `partial` (from `functools`) on [MSE](https://scikit-learn.org/stable/modules/generated/sklearn.metrics.mean_squared_error.html), will work (just set "squared=False")

**Review by IanAtCredo — CHANGES_REQUESTED** [review]
Few small changes

File: credoai/metrics/metric_constants.py (hunk 2 above)
Comment (amrasekh) [reply credoai/metrics/metric_constants.py:72]: Added

```
    'mean_absolute_error': ['MAE'],
    'mean_squared_error': ['MSE'],
    'root_mean_squared_error': ['RMSE'],
    'r2_score': ['r_squared']
```

File: credoai/metrics/metric_constants.py (hunk 3 above)
Comment (amrasekh) [reply credoai/metrics/metric_constants.py:76]: Added
`'root_mean_squared_error': partial(sk_metrics.mean_squared_error, squared=False)`

File: credoai/metrics/metric_constants.py [hunk 4]
```diff
@@ -53,7 +54,11 @@
     'demographic_parity_ratio': ['disparate_impact'], 
     'average_odds_difference': ['average_odds'], 
     'equal_opportunity_difference': ['equal_opportunity'], 
-    'equalized_odds_difference': ['equalized_odds']
+    'equalized_odds_difference': ['equalized_odds'],
+    'mean_absolute_error': ['MAE'],
+    'mean_squared_error': ['MSE'],
+    'root_mean_squared_error': ['RMSE'],
+    'r2_score': ['r_squared']
```
Comment (IanAtCredo) [inline credoai/metrics/metric_constants.py:61]:
```suggestion
    'r2_score': ['r_squared', 'r2']
```

File: credoai/metrics/metric_constants.py [hunk 5]
```diff
@@ -53,7 +54,11 @@
     'demographic_parity_ratio': ['disparate_impact'], 
     'average_odds_difference': ['average_odds'], 
     'equal_opportunity_difference': ['equal_opportunity'], 
-    'equalized_odds_difference': ['equalized_odds']
+    'equalized_odds_difference': ['equalized_odds'],
+    'mean_absolute_error': ['MAE'],
+    'mean_squared_error': ['MSE'],
```
Comment (IanAtCredo) [inline credoai/metrics/metric_constants.py:59]:
```suggestion
    'mean_squared_error': ['MSE', 'MSD', 'mean_squared_deviation'],
```

**Review by IanAtCredo — APPROVED** [review]
ready to merge after you accept suggestions
